x-pack/platform/plugins/shared/cases/server/attachment_framework/types.ts

```diff
@@ -32,11 +39,40 @@ export interface ExternalReferenceAttachmentType {
   schemaValidator?: (data: unknown) => void;
 }
 
+export interface SuggestionType<TPayload extends {} = {}> {
+  /* Unique identifier for the suggestion type */
+  id: string;
+  /* Unique identifier for the type of attachment the suggestion is for */
+  attachmentId: string;
```

Comment:
I'd include the word "type" here to make it clear that it refers to the type of the attachment and not to the id.

```suggestion
  attachmentTypeId: string;
```